host/volume/zfs/zfs.go

```diff
@@ -452,11 +457,8 @@ func (b *Provider) RestoreVolumeState(volInfo *volume.Info, data json.RawMessage
 		dataset:   dataset,
 		basemount: record.Basemount,
 	}
-	// zfs should have already remounted filesystems; special remount case for snapshots
-	if v.IsSnapshot() {
-		if err := b.mountSnapshot(v); err != nil {
-			return nil, err
-		}
+	if err := b.mountDataset(v); err != nil {
```

Comment:
It should be idempotent, atleast I am sure of the `zfs mount` command, I will test `mount -t zfs` to test that too but that code-path was already running. I think that probably wouldn't have mattered right now though as I don't think we are using the snapshotting features?